pkg/ds/fields/fields.go

```diff
@@ -0,0 +1,117 @@
+package fields
+
+import (
+	"encoding/json"
+
+	"k8s.io/kubernetes/pkg/labels"
+
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/types"
+)
+
+// ID is a named type alias for Daemon Set IDs
+type ID string
+
+// Cluster name is where the daemon set lives on
+type ClusterName string
+
+// DaemonSet holds the runtime state of a Daemon Set as saved in Consul.
+type DaemonSet struct {
+	// UUID for this controller
+	ID ID
+
+	// When disabled, this controller will not make any scheduling changes
+	Disabled bool
+
+	// The pod manifest that should be scheduled on nodes
+	Manifest pods.Manifest
+
+	// Minimum health for nodes when scheduling
+	MinHealth int
+
+	// Daemon Set's environment name
+	Name ClusterName
```

Comment:
@anthonybishopric do you not think that this should be on the pod cluster level?